app/assets/javascripts/discourse/app/lib/hashtag-decorator.js

```diff
@@ -23,6 +25,16 @@ export function fetchUnseenHashtagsInContext(
       response[type].forEach((item) => {
         seenHashtags[type][item.ref] = seenHashtags[type][item.ref] || item;
       });
+
+      if (type === "category") {
+        const categoryIds = response[type].map((item) => item.id);
+        // The only relevant side effect is creation of CSS styles that are
+        // immediately rendered to the user
+        Category.asyncFindByIds(categoryIds).then((categories) => {
```

Comment:
Lost the response payload comment, but I agree we should reduce this with a lightweight response...all we need is this for each category:

* id
* color
* text_color
* parent_category (which also has id, color, text_color)

And we are currently loading everything for categories:

![image](https://github.com/discourse/discourse/assets/920448/de9a2a28-9383-4b06-b92d-2736fbb4881a)